mujoco_warp/_src/forward.py

```diff
@@ -536,7 +536,8 @@ def fwd_position(m: Model, d: Data, factorize: bool = True):
   smooth.tendon_armature(m, d)
   if factorize:
     smooth.factor_m(m, d)
-  collision_driver.collision(m, d)
+  if run_collision_detection:
```

Comment:
Here you can do:

```py
if m.opt.run_collision_detection:
  ...
```

Then you can remove the `run_collision_detection` param that you added.

Similarly we will remove `factorize` param in a later cleanup.